### PR TITLE
fix(web): polyline accurately updates, deep cloned objects

### LINKED ISSUE #120: Map route should auto update when locations are changed
### Expected Behavior

Get new directions data via api when event locations change, removing old route and adding new

### Current Behavior

Doesn't get new directions unless page refreshed, [`componentDidUpdate()`](https://github.com/drklee3/YOUtinerary/blob/master/packages/web/src/components/Plan.tsx#L127) in `Plan.tsx` seems to give the current state as prevstate.  Could possibly be issues with state being array and some reference comparisons. 

This also happens:

image

* Reordering events refreshes the route with the new one.
* Changing an event location **does not** get a new route (before/after waypoints logging shows the new location for both)
* Adding a new event and adding a location **does not** get a new route.

### PR DESCRIPTION
**Pull request by pbebla**
### Description

The polyline accurately updates, fixing cases where there's less than 2 markers. Adding/Removing events auto update line correctly.

### Tested

Added test markers to test and it works.

### Related issues

* Fixes #120

### CODE REVIEW

**Comment by drklee3** [comment]
Seems like it works properly, good work.

There are a handful of tests failing though, could you look into those and fix them?

**Comment by pbebla** [comment]
_.cloneDeep(events) in editEventField() breaks the test cases, but I don't understand why. The function needs it because you modify a value in a nested array, so shallow clone won't copy by value, it'll by copy by reference everything past the top level.

**Comment by pbebla** [comment]
Oh! I think it's because in the test script, ur checking the value of plan[1].name when the modified array hasn't been assigned to plan yet. Before since it was copy by reference, calling editEventField changed both arrays at the same time. So I think there's need to be an extra step in the test before the comparison.

**Comment by drklee3** [comment]
Yeah I see a lot of the tests are relying on the functions to modify the array in place, could you update the tests to reflect your changes that make them immutable?

**Comment by pbebla** [comment]
for my search_test, it says I'm missing a valid API key, client ID, or secret. I don't know why.

File: packages/web/src/itinerary/PlanHelpers.test.ts [hunk 1]
```diff
@@ -83,27 +83,27 @@ describe("plan", () => {
     it("should correctly edit event names", () => {
         const plan = newPlan();
 
-        editEventName(plan, 1, "my name was changed");
+        const test = editEventName(plan, 1, "my name was changed");
```
Comment (drklee3) [inline packages/web/src/itinerary/PlanHelpers.test.ts:86]: :+1:

**Review by drklee3 — APPROVED** [review]
LGTM! Good work

**Pull request merged**